### PR TITLE
0.3 docs update

### PR DESCRIPTION
**Pull request by jackdotink**

### CODE REVIEW

File: docs/config/options.md [hunk 1]
```diff
@@ -34,11 +50,17 @@ The casing option defines the casing of the functions generated by Zap.
...
+## `write_checks`
 
-## Writechecks
+This option determines if Zap should check types when writing data to the network. This is useful for development and debugging, but can see some performance hits, and should be disabled in production.
 
-The writechecks option determines if conditional constraints should be valided on the just the receiver (`false`) or on both the sender and receiver (`true`).
+::: danger
+Zap only checks types that cannot be statically checked by Luau or TypeScript.
+
+For example, Zap will not check if a `string (20)` is a string, but it will check that the string is 20 characters long.
```
Comment (sasial-dev) [inline docs/config/options.md:62]: As @nezuo suggested - we should explain how buffers don't require us to do all checks runtime

File: docs/config/types.md [hunk 2]
```diff
@@ -1,105 +1,180 @@
...
+| `i8`  | -128           | 127           |
+| `i16` | -32,768        | 32,767        |
+| `i32` | -2,147,483,648 | 2,147,483,647 |
 
-<CodeBlock code="type Health = u8 (0..100)" />
+### Unsigned Integers
 
-This works because limits are **inclusive**, and will include 0 and 100.
-## Strings
+Unsigned integers are positive integers that cannot be negative.
```
Comment (sasial-dev) [inline docs/config/types.md:71]:
```suggestion
Unsigned integers are positive integers.
```

File: docs/config/types.md [hunk 3]
```diff
@@ -1,105 +1,180 @@
...
 
-Enums are values seperated by a comma (`,`) inside brackets (`()`). For example:
-<CodeBlock code="type RoundStatus = ( Playing, Intermission )" />
+Zap has two types of enums, unit enums and tagged enums.
+
+### Unit Enums
+
+Unit enums are used to represent a set of possible values. They are defined using the `enum` keyword, followed by a set of possible string values. For example, a unit enum representing the status of a round would look like:
+
+<CodeBlock code='enum { Starting, Playing, Intermission }' />
```
Comment (sasial-dev) [inline docs/config/types.md:131]:
```suggestion
<CodeBlock code='enum { "Starting", "Playing", "Intermission" }' />
```

File: docs/config/types.md [hunk 4]
```diff
@@ -1,105 +1,180 @@
...
+
+This code would create the Luau type:
```
Comment (sasial-dev) [inline docs/config/types.md:133]:
```suggestion
This code would then create the Luau type:
```

File: docs/intro/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+```bash
+aftman add red-blox/zap
+```
+
+Alternatively you can use the [web playground](https://zap.redblox.dev/playground) and move the generated files into your project.
+
+## Writing Your First Network Description
+
+Zap's IDL (Interface Description Language) is very simple, and you'll learn more about it further in this guide. For now, here's a simple example:
```
Comment (sasial-dev) [inline docs/intro/getting-started.md:46]: I wonder if we can use a simpler term than IDL?

File: docs/intro/what-is-zap.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# What is Zap?
+
+Zap is a tool to generate highly efficient networking code for Roblox. It has no compromises on performance or developer experience. Zap takes a description of your data and event, and generates code specific to your game.
+
+<div class="custom-block tip" style="padding-top: 8px">
```
Comment (sasial-dev) [inline docs/intro/what-is-zap.md:5]: Why did you add a custom tip? It may look better if you did an info tip with a custom title

File: docs/usage/events.md [hunk 7]
```diff
@@ -1,42 +1,154 @@
...
+    data: {
+        baz: boolean,
+    },
+}`
+</script>
+
 # Using Zap
 
-All of Zap's output code is **fully typed**, and will pickup if you input the wrong type.
+Zap's generated API is fully typesafe. While Zap will not throw errors for invalid usage, your static analysis tooling will.
```
Comment (sasial-dev) [inline docs/usage/events.md:24]: zap will not throw errors unless there is a range specified

File: docs/usage/events.md [hunk 8]
```diff
@@ -1,42 +1,154 @@
...
+
+- If you're using [vscode](https://code.visualstudio.com/), we recommend installing the [Luau-LSP](https://marketplace.visualstudio.com/items?itemName=JohnnyMorganz.luau-lsp) extension.
```
Comment (sasial-dev) [inline docs/usage/events.md:26]:
```suggestion
- If you're using [VSCode](https://code.visualstudio.com/), we recommend installing the [Luau-LSP](https://marketplace.visualstudio.com/items?itemName=JohnnyMorganz.luau-lsp) extension.
```

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Just a heads up -> don't commit these suggestions individually - just do it all in one go.

File: docs/intro/getting-started.md (hunk 5 above)
Comment (jackdotink) [reply docs/intro/getting-started.md:46]: IDL is the actual term, and some people *cough Marcus cough* prefer using the correct term

File: docs/intro/what-is-zap.md (hunk 6 above)
Comment (jackdotink) [reply docs/intro/what-is-zap.md:5]: I didn't really want to give it a title, it looked kinda clunky.

File: docs/config/options.md (hunk 1 above)
Comment (jackdotink) [reply docs/config/options.md:62]: This is a good idea, but I'm not sure where to put it in the docs

File: docs/config/events.md [hunk 9]
```diff
@@ -1,72 +1,58 @@
...
-
-Type determines if the event should be considered reliable or unreliable. Reliable events are ordered and guaranteed to deliver, while unreliable events are unordered and are not guaranteed to deliver.
-
-### Default
-
-`Reliable`
+# Events
 
-### Options
+Events are the primary method of communicating between the client and the server. Events are also what is exposed to the developer from Zap's generated API.
```
Comment (sasial-dev) [inline docs/config/events.md:15]:
```suggestion
Events are the primary method of communication between the client and the server. Events are also what is exposed to the developer from Zap's generated API.
```

File: docs/config/options.md [hunk 10]
```diff
@@ -1,26 +1,42 @@
...
+
+This option changes the casing of the API generated by Zap.
 
-## Casing
+Using the `FireAll` function as an example:
 
-The casing option defines the casing of the functions generated by Zap.
+- `PascalCase`: `FireAll`
+- `camelCase`: `fireAll`
+- `snake_case`: `fire_all`
```
Comment (sasial-dev) [inline docs/config/options.md:35]: this should be a table

File: docs/config/options.md [hunk 11]
```diff
@@ -34,11 +50,17 @@ The casing option defines the casing of the functions generated by Zap.
 
 ### Example
 
-<CodeBlock code="opt casing = PascalCase" />
+<CodeBlock code="opt casing = camelCase" />
```
Comment (sasial-dev) [inline docs/config/options.md:53]: remove str from lalrpop and make it a literal

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Some more...

**Review by sasial-dev — APPROVED** [review]
LGTM!
